### ISSUE TITLE
Errors when fetching image caches are silenced sometimes

### ISSUE THREAD

**Issue by tailhook** [tailhook/vagga#363]
So I had small amount of free disk space, and this triggered the following error:
```
vagga _build container                          
Connecting to image-cache.internal (192.168.1.2:80)
d281d46f-container-56cf   9% |*********                                                                                      | 26780k  0:00:27 ETA
ERROR:vagga::builder: Error building container "container": step SubConfig { source: Container("docker-parser"), path: "vagga.yaml", container: "do
cker-smart", cache: None } failed: sub-step Ubuntu("trusty") failed: Error writing "/vagga/root/usr/bin/sg": File exists (os error 17)
ERROR:vagga::wrapper: Error executing _build: Builder exited with code 1
Command <Command "/proc/self/exe" ("vagga_wrapper") "_build" "container"; environ: {"TERM"="screen","_VAGGA_HOME"="/home/pc","RUST_LOG"="warn",}; uid_map=[UidMap { inside_uid: 0, outside_uid: 1000, count: 1 }, UidMap { inside_uid: 1, outside_uid: 100000, count: 65535 }]; gid_map=[GidMap { inside_gid: 0, outside_gid: 100, count: 1 }, GidMap { inside_gid: 1, outside_gid: 100000, count: 65535 }]> exited with code 124
```

On second run it doesn't download the file any more. But shows the same error.

I.e. wget silences error and we consider archive fine, while it's actually broken. On the other hand I I just use `Ctrl+C` when downloading, it deletes the file and next run starts over:

```            
Connecting to image-cache.internal (192.168.1.2:80)
d281d46f-container-56cf  59% |********************************************************                                       |   159M  0:00:10 ETA^C
ERROR:vagga::builder::download: Error downloading "http://image-cache.internal/containers/container-56cf4760.tar.gz", wget Signaled(2, false)   
```

**Comment by anti-social** [comment]
Can't reproduce `wget` behavior:
```
/tmp # busybox wget http://mirror.datacenter.by/pub/archlinux/iso/2016.11.01/archlinux-2016.11.01-dual.iso
Connecting to mirror.datacenter.by (178.124.134.106:80)
archlinux-2016.11.01  12% |***********                                                                                   |    99M  0:08:04 ETAwget: short write
/tmp # echo $?
1
```

**Comment by tailhook** [comment]
Hm, same here:
```
> vagga _build container
Connecting to image-cache.internal (192.168.1.2:80)
wget: short write
ERROR:vagga::builder::download: Error downloading "http://image-cache.internal/containers/container-56cf4760.tar.gz", wget Exited(1)
(1/1) Installing alpine-keys (1.1-r0)
OK: 0 MiB in 1 packages
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
```

But I could reproduce it multiple times on a different machine. I'll try again tomorrow.

**Comment by tailhook** [comment]
Well, it turns out I can't reliably reproduce it anymore.

**Comment by tailhook** [comment]
It might be related to [this todo comment](https://github.com/tailhook/vagga/blob/master/src/builder/mod.rs#L106)

**Comment by tailhook** [comment]
So this is how it works, I presume:

1. Image is downloaded and fine (progressbar isn't good indicator that download failed AFAICS)
2. Then we try to unpack image and fail when unpacking
3. Image unpacking error was silenced, because we want to rebuild from scratch
4. But we don't clean directory before rebuilding
5. So when unpacking ubuntu image on top of original half-unpacked file from cache we got error

Commit 9c7b875 logs the original error and cleans dir on failure.

**Comment by tailhook** [comment]
Okay, it's now better, but not fully fixed:
```
ERROR:vagga::builder: Error when unpacking image: Error unpacking image http://cache.internal/some.tar.gz: Error unpacking "/vagga/cache/downloads/2fc402c1-some.tar.gz" -> "/vagga/root/./usr/bin/python3.4": No space left on device (os error 28). Will clean and build it locally...
ERROR:vagga::builder: Error building container "container": recursive bind mount "/dev" -> "/vagga/root/dev": No such file or directory (os error 2) (source: exists, target: missing, mapped-root)
ERROR:vagga::wrapper: Error executing _build: Builder exited with code 1
Command <Command "/proc/self/exe" "__wrapper__" "_build" "container"; environ[3]; uid_map=[UidMap { inside_uid: 0, outside_uid: 1000, count: 1 }, UidMap { inside_uid: 1, outside_uid: 100000, count: 65535 }]; gid_map=[GidMap { inside_gid: 0, outside_gid: 100, count: 1 }, GidMap { inside_gid: 1, outside_gid: 100000, count: 65535 }]> exited with code 124
```

**Comment by tailhook** [comment]
Fixed in 0.7.2.
